### ISSUE TITLE
prepare refactoring: add more tooling in development workflows

### ISSUE THREAD

**Issue by stweise** [gtaubman/doneyet#40]
I would like to use these tools to: 
* avoid having coding style troubles with each other (I am lazy and might make mistakes a formatter will spot)
* detect & fix easily avoidable coding issues (before and during refactoring)

Here are my suggestions:

static code analyzer: http://cppcheck.sourceforge.net/
linter: cpplint.py from https://github.com/google/styleguide/tree/gh-pages/cpplint
automatic formatting: astyle from http://astyle.sourceforge.net/

I have already started using them when working on my repo: please see master of https://github.com/stweise/doneyet
parameters chosen are documented in file called FORMATTING

**Comment by gtaubman** [comment]
For auto formatting, do you object to clang-format with Google style?

**Comment by stweise** [comment]
Sounds good. Tested it and it looks nice.

Used: clang-format -i  --style=google *.cc *.h
I noticed it does not comply to cpplint.py, which is frustrating. That is something i worked out in my melange of options/filters using **astyle** and **cpplint.py**.
I would rather use one tool than try to fiddle with two tools all the time, so going for clang-format would be ok for me.

I would like to test it a little and later put it into the **pre-commit** hook in git.
